src/components/organisms/AdminVenueView/components/SearchField/SearchField.tsx

```diff
@@ -0,0 +1,64 @@
+import { debounce } from "lodash";
+import React, { useState, useMemo } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTimesCircle } from "@fortawesome/free-solid-svg-icons/faTimesCircle";
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
+
+import "./SearchField.scss";
+
+type onChange = (value: string) => void;
+type CustomInputProps = "value" | "onChange"; // we override the types of these standard props
+
+export interface SearchUsersProps
+  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, CustomInputProps> {
+  onChange?: onChange;
+  value?: string;
+  placeholder?: string;
+  debounceTime?: number;
+}
+
+export const SearchField: React.FC<SearchUsersProps> = ({
+  onChange,
+  value = "",
+  debounceTime = 100,
+  ...props
+}) => {
+  const [text, setText] = useState(value);
+  const debouncedHandleChange: onChange = useMemo(
+    () => debounce((value) => onChange?.(value), debounceTime),
+    [debounceTime, onChange]
+  );
+
+  const cn = classNames({
```

Comment:
let's rename it into smth more clear, like SearchFieldClassnames or smth like that

Reply:
LOL, It's like saying
> lets rename "smth" into something more clear, like "aThingThatJustIs" or something like that.

Does a name `cn` have a problem being unclear when it's being assigned a value from a function named `classNames`?

I see `cn` as an equivalent example of having `for( i =0; ... )`, a clear meaning in the context that doesn't warrant unnecessary attention by being overly verbose.

Reply:
Some of our files have multiple of these and it makes sense to name them with more element specific names. I'll change it to `searchFieldClassNames` to make it more obvious that these class names are towards that BEM element.

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1705/commits/3ca884136419f135a2893c7e0fc2ee4af1d3301b